### ISSUE TITLE
Polluting global namespace

### ISSUE THREAD

**Issue by MorsMortium** [Wiladams/LJIT2Khronos#1]
I don't know, if it's intended, but all the constants are global. It would be great, if they could be accessed the same way as functions.

**Comment by Wiladams** [comment]
sins of the past.  This is a fairly ancient binding.  These days, I have a few ways of dealing with constants and enums.

Some times I will put them into an "enum" thing, which actually serves as a database so you can lookup names of things from their constants, as well it can generate cdefs if you want to access things through ffi.C.

What's currently in this library is an older sort of, just make it look and feel like what a C programmer would experience without much fuss, but then I threw in some conveniences for looking things up.  

Bottom line, this thing needs revisiting.  If I do that, I will 'do the right things'.

**Comment by MorsMortium** [comment]
I would very much like it, I've been using glew for opengl, for a little project, but yours works better, on multiple platforms, plus the fact, that it's native lua code, makes it better (less dependency, faster)

**Comment by Wiladams** [comment]
Ok.  Which version of opengl do you target?  Ill start there.

**Comment by MorsMortium** [comment]
I've been using 3.3, but I just realized after reading through the code, that it's only compatible with windows right now. Are you planning on making it cross-platform?

**Comment by Wiladams** [comment]
Ive done this on linux before as well, including raspberry pi.

If theres some interest, ill do it.  Ive done quite a lot of work on linux ui stack, so, totally possible.

Only the absolute binding to windowing environment is of concern.

Otherwise, there are other opengl bindings since.  Which opengl profile do you seek?  3.x or 4.x?
________________________________

**Comment by MorsMortium** [comment]
I've been using 3.x, but I'm looking for a general solution, which can fetch any version, at runtime. For windowing, I am using SDL2, which is cross platform, and has SDL_GL_GetProcAddress

**Comment by MorsMortium** [comment]
Bump
